Sort the tags and tags' categories case insensitive

lgalis opened the pull request:
The tags and tag categories are sorted case sensitive in the dropdown list. They should be sorted case-insensitive

https://bugzilla.redhat.com/show_bug.cgi?id=1238580

lgalis commented:
@miq-bot add_label ui, bug

lgalis commented:
Before:

![screenshot from 2016-06-15 15-29-05](https://cloud.githubusercontent.com/assets/12769982/16094857/163beb38-3310-11e6-8928-041f5af4a645.png)
![screenshot from 2016-06-15 15-29-18](https://cloud.githubusercontent.com/assets/12769982/16094858/163c4362-3310-11e6-97a1-74469288d6c6.png)


After:

![screenshot from 2016-06-15 15-39-10](https://cloud.githubusercontent.com/assets/12769982/16094874/239a0350-3310-11e6-82b9-23af767c43ad.png)
![screenshot from 2016-06-15 15-39-22](https://cloud.githubusercontent.com/assets/12769982/16094873/2393844e-3310-11e6-9cac-52a81f601f8d.png)

lgalis commented:
@dclarizio - please review

dclarizio commented:
@lgalis On the second example screen, showing the numbers and text choices, somehow the `<Select...` choice is in the middle of the list between the number choices and the text choices.  It should be on the top of the list, prepended after the list gets sorted.

lgalis commented:
@dclarizio - the <select>  was part of the list for a long time - may not have been noticed as tags may not start with number in most cases. I fixed that now and updated the second 'After' screenshot.

chessbyte commented:
Darga backport details:
```sh
commit c5da822b5f72964277a11032c104b1dfa9d5b658
Author: Dan Clarizio <dclarizi@redhat.com>
Date:   Thu Jun 16 09:32:35 2016 -0700

    Merge pull request #9234 from lgalis/sort_company_tags_case_insensitive
    
    Sort the tags and tags' categories case insensitive
    (cherry picked from commit 229eafa61a0774a1d4a31a89cb2f67563eb992ec)
```
